### ISSUE TITLE
Error : range start index 3602996316 out of range for slice of length 1795

### ISSUE THREAD

**Issue by iKineticate** [lilopkins/lnk-rs#25]
```rust
dbg!(lnk::ShellLink::open(r"C:\Users\11593\Desktop\GitHub Desktop.lnk").unwrap());
```

Error Message:
thread 'main' panicked at lnk-rs\src\linkinfo.rs:355:46:
range start index 3602996316 out of range for slice of length 1795

The LNK file:
[GitHub Desktop.zip](https://github.com/lilopkins/lnk-rs/files/15263587/GitHub.Desktop.zip)

Wish you good health.❤️

**Comment by isma-xnm** [comment]
Encountered this as well and made a PR fix that prevents it from panicing in. @iKineticate Do you know if the file you attached is actually corrupted, or if it's expected to be read correctly? Since the length is way larger than the buffer, the library still cannot parse the file correctly, but maybe that is correct?

**Comment by iKineticate** [comment]
> Encountered this as well and made a PR fix that prevents it from panicing in. @iKineticate Do you know if the file you attached is actually corrupted, or if it's expected to be read correctly? Since the length is way larger than the buffer, the library still cannot parse the file correctly, but maybe that is correct?

I haven't made any modifications to the shortcut. After a certain day, all newly created or updated shortcuts can no longer be parsed by the library.However, when using the WinAPI directly, the shortcuts can be parsed correctly.

**Comment by isma-xnm** [comment]
Yeah, I don't exactly understand why, but your `.lnk` contains a network share path that isn't used (as the link target is on the same computer). The error is in the parsing of that network path (and for us, only happens when we actually have shortcuts that point to a network shared path). In any case, fixing the off-by-one error in the parsing should solve it for both our cases.
